### PR TITLE
Add Extending Bitcoin Provider API apprenticeship

### PR DESCRIPTION
**Pull request by mikeosborne**

### CODE REVIEW

File: content/grants/apprenticeship-extending-bitcoin.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+- Knows some functional programming and is interested in learning Hoon
+
+- Has done backend webdev or otherwise coded against REST APIs
+
+- Has experience with or is interested in bitcoin development
+
+## Time Estimate
+
+For someone with zero familiarity with urbit and bitcoin APIs, and working 10 hours a week or less, I would estimate 3-4 weeks.
```
Comment (ghost) [inline content/grants/apprenticeship-extending-bitcoin.md:41]: @mikeosborne did we want to change this time estimate to 2 months?

File: content/grants/apprenticeship-extending-bitcoin.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Deliverables
+
+- Add support for regtest to the bitcoin codebase
```
Comment (ghost) [inline content/grants/apprenticeship-extending-bitcoin.md:45]: Looking at this again with fresh eyes, would it be helpful for me to expand on this? It touches both the electrs proxy and the urbit bitcoin side.

Comment (mikeosborne) [reply content/grants/apprenticeship-extending-bitcoin.md:45]: Yeah - more detail would be good

File: content/grants/apprenticeship-extending-bitcoin.md (hunk 1 above)
Comment (mikeosborne) [reply content/grants/apprenticeship-extending-bitcoin.md:41]: It kind of depends on whether you are willing to take on someone without as much experience for your prerequisites. We should probably change it to 4-8 weeks DOE.

File: content/grants/apprenticeship-extending-bitcoin.md (hunk 2 above)
Comment (ghost) [reply content/grants/apprenticeship-extending-bitcoin.md:45]: The task is not too huge, I might break it down as follows:

- add %regtest to `network` type defined in `/=bitcoin=/sur/bitcoin.hoon`
- modify functions in `/=bitcoin=/lib/` to handle this updated type appropriately
- update server.js in RPC repo to support new network configuration
- run some "functional tests", sending and receiving transactions using regtest configuration

**Review by jalehman — APPROVED** [review]

**Pull request merged**